packages/framework/src/Foundation/Providers/NavigationServiceProvider.php

```diff
@@ -21,7 +21,22 @@ public function register(): void
 
         $this->app->alias(NavigationManager::class, 'navigation');
 
+        // Initially bind these to null, so that the container doesn't try to return empty menus before the kernel has booted.
```

Comment:
Nvm, I only want lazy booting because it makes unit tests easier. But the DX for plugin developers is more important, because trying to add a menu too early will lead to issues. It must be done after kernel boot.